client/my-sites/site-settings/form-general.jsx

```diff
@@ -127,35 +120,68 @@ const FormGeneral = React.createClass( {
 		return this.recordEventOnce.bind( this, key, eventAction );
 	},
 
+	setDirtyField( key ) {
+		const newState = {};
+		const dirtyFields = this.state.dirtyFields || [];
+		if ( dirtyFields.indexOf( key ) === -1 ) {
+			newState.dirtyFields = [ ...dirtyFields, key ];
+		}
+		this.setState( newState );
```

Comment:
Also just noticed that instead of doing  `const dirtyFields = this.state.dirtyFields || [];` we could just initialize the state with an empty array in `getInitialState` and then change this to 
`const { dirtyFields } = this.state;`